src/frontend/src/scheduler/worker_node_manager.rs

```diff
@@ -140,63 +133,200 @@ impl WorkerNodeManager {
         Ok(workers)
     }
 
-    pub fn get_fragment_mapping(&self, fragment_id: &FragmentId) -> Option<ParallelUnitMapping> {
+    pub fn get_streaming_fragment_mapping(
+        &self,
+        fragment_id: &FragmentId,
+    ) -> Option<ParallelUnitMapping> {
         self.inner
             .read()
             .unwrap()
-            .fragment_vnode_mapping
+            .streaming_fragment_vnode_mapping
             .get(fragment_id)
             .cloned()
     }
 
-    pub fn insert_fragment_mapping(
+    pub fn insert_streaming_fragment_mapping(
         &self,
         fragment_id: FragmentId,
         vnode_mapping: ParallelUnitMapping,
     ) {
         self.inner
             .write()
             .unwrap()
-            .fragment_vnode_mapping
+            .streaming_fragment_vnode_mapping
             .try_insert(fragment_id, vnode_mapping)
             .unwrap();
     }
 
-    pub fn update_fragment_mapping(
+    pub fn update_streaming_fragment_mapping(
         &self,
         fragment_id: FragmentId,
         vnode_mapping: ParallelUnitMapping,
     ) {
-        self.inner
-            .write()
-            .unwrap()
-            .fragment_vnode_mapping
+        let mut guard = self.inner.write().unwrap();
+        guard
+            .streaming_fragment_vnode_mapping
             .insert(fragment_id, vnode_mapping)
             .unwrap();
+        // TODO #8940: re-calculate serving schedule to keep locality with best efforts
+        guard.serving_fragment_vnode_mapping.clear();
```

Comment:
I think `clear()` could affect too many fragments. How about removing the corresponding `fragment_id`?

Reply:
Ideally we can implement a rebalancing method similar to [rebalance_actor_vnode](https://github.com/risingwavelabs/risingwave/blob/cce2d05089ec5872c417944b28b1eaa84e4f99a3/src/meta/src/stream/scale.rs#L126).
I simply `clear` here to avoid too large PR.

Reply:
Thanks, it's a mistake.
Also refactor to round-robin when scheduling serving PUs. PTAL.